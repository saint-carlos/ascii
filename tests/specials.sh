#!/bin/bash

. $TESTDIR/common.sh

str[0]="100 120 40"
str[1]=`seq 32 126`
str[2]=`seq 0 255`
str[3]=""
str[4]=`seq -1000 1000`
str[5]=`jot -r 600 -275 550`

special_to_char_sed()
{
	tail -n +2 tests/specials.out  | tr '\t' ':' | awk -F: "$1"
}

short_to_char_sed()
{
	special_to_char_sed '{ printf("s,\\[%s\\],\\x%x,g;", $3, $1); }'
}

long_to_char_sed()
{
	special_to_char_sed '{ printf("s,\\[\"%s\\\"],\\x%x,g;", $4, $1); }'
}

special_to_char()
{
	local SCRIPT="$($1)"
	sed "$SCRIPT"
}

FILE="$TMPDIR/$$"

$ascii -l > $FILE.actual
diff -u tests/specials.out $FILE.actual || exit 1
$ascii -t > $FILE.actual
diff -u tests/all.out $FILE.actual || exit 1

for (( i=0 ; i < ${#str[@]} ; i++))
do
	echo "i=" $i

	echo case=an
	echo "${str[i]}" | $ascii -an > $FILE.actual
	echo "${str[i]}" | dummy_ascii true | to_chars > $FILE.expected
	vcmp $FILE.expected $FILE.actual || exit 1

	echo case=ans
	echo "${str[i]}" | $ascii -ans | special_to_char short_to_char_sed > $FILE.actual
	echo "${str[i]}" | dummy_ascii true | to_chars > $FILE.expected
	vcmp $FILE.expected $FILE.actual || exit 1

	echo case=anS
	echo "${str[i]}" | $ascii -anS | special_to_char long_to_char_sed > $FILE.actual
	echo "${str[i]}" | dummy_ascii true | to_chars > $FILE.expected
	vcmp $FILE.expected $FILE.actual || exit 1
done
exit 0
