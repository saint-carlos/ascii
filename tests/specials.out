Dec	Hex	Short	Long
  0	00	NUL	null
  1	01	SOH	start of heading
  2	02	STX	start of text
  3	03	ETX	end of text
  4	04	EOT	end of transmit
  5	05	ENQ	enquiry
  6	06	ACK	positive acknowledge
  7	07	BEL	bell
  8	08	BS	backspace
  9	09	HT	horizontal tab
 10	0a	LF	line feed
 11	0b	VT	vertical tab
 12	0c	FF	form feed
 13	0d	CR	carriage return
 14	0e	SO	shift out
 15	0f	SI	shift in
 16	10	DLW	data link escape
 17	11	DC1	device control 1
 18	12	DC2	device control 2
 19	13	DC3	device control 3
 20	14	DC4	device control 4
 21	15	NAK	negative acknowledge
 22	16	SYM	synchronous idle
 23	17	ETB	end of transmit block
 24	18	CAN	cancel
 25	19	EM	end of medium
 26	1a	SUB	substitute
 27	1b	ESC	escape
 28	1c	PS	file separator
 29	1d	GS	group separator
 30	1e	RS	record separator
 31	1f	US	unit separator
127	7f	DEL	delete
